app/models/user.rb

```diff
@@ -1,4 +1,6 @@
 class User < ActiveRecord::Base
+  attr_accessor :asserted_attributes
```

Comment:
Reek is complaining about this because it treats public accessor methods as a code smell. Since this is a transient attribute that is only used in the context of SAML auth, it does not belong in the User class. Instead, I would create a new class that returns an object that can be passed to the SamlIdp library.

```ruby
class UserWithAssertedAttributes
  def initialize(user)
    @user = user
  end

  def call(attrs)
    self.asserted_attributes = attrs
  end

  delegate :email, :mobile, to: :user

  attr_reader :user, :asserted_attributes

  private

  attr_writer :asserted_attributes
end
```
Usage:
```ruby
user = UserWithAssertedAttributes.new(User.find(1))
user.call({ foo: 'bar' })

user.asserted_attributes
=> {:foo=>"bar"}

user.email
=> "test1@test.com"
```
If the SamlIdp gem needs to know about any attributes other than the asserted attributes, they can be delegated, like `email` and `mobile` were above.

By making the `attr_writer` private, it prevents outsiders from directly setting the attributes, and what we return is an object that only responds to messages that the SamlIdp library needs to know about.